src/odm-collection.hpp

```diff
@@ -0,0 +1,656 @@
+        const mongocxx::options::insert& options = mongocxx::options::insert());
+
+    ///
+    /// Inserts multiple serializable objects into the collection.
+    ///
+    /// TODO how to deal w/ identifiers
+    /// If any of the documents are missing /// identifiers the driver will generate them.
+    ///
+    /// @warning This method uses the bulk insert command to execute the insertion as opposed to
+    /// the legacy OP_INSERT wire protocol message. As a result, using this method to insert many
+    /// documents on MongoDB < 2.6 will be slow.
+    ///
+    /// @tparam object_iterator_type
+    ///   The iterator type. Must meet the requirements for the input iterator concept with a value
+    ///   type that is a serializable object.
+    ///
+    /// @param begin
+    ///   Iterator pointing to the first document to be inserted.
+    /// @param end
+    ///   Iterator pointing to the end of the documents to be inserted.
+    /// @param options
+    ///   Optional arguments, see mongocxx::options::insert.
+    ///
+    /// @return The result of attempting to performing the insert.
+    /// @throws exception::write if the operation fails.
+    ///
+    /// TODO: document DocumentViewIterator concept or static assert
+    template <typename object_iterator_type>
+    mongocxx::stdx::optional<mongocxx::result::insert_many> insert_many(
+        object_iterator_type begin, object_iterator_type end,
+        const mongocxx::options::insert& options = mongocxx::options::insert());
+
+    ///
+    /// Replaces a single document matching the provided filter in this collection.
+    ///
+    /// @param filter
+    ///   Document representing the match criteria.
+    /// @param replacement
+    ///   The replacement document, as a serializable object.
+    /// @param options
+    ///   Optional arguments, see mongocxx::options::update.
+    ///
+    /// @return The result of attempting to replace a document.
+    /// @throws exception::write if the operation fails.
+    ///
+    /// @see http://docs.mongodb.org/manual/reference/command/update/
+    ///
+    mongocxx::stdx::optional<mongocxx::result::replace_one> replace_one(
+        bsoncxx::document::view_or_value filter, const T& replacement,
+        const mongocxx::options::update& options = mongocxx::options::update());
+
+    ///
+    /// Replaces a single document matching the provided serializable object in this collection.
+    ///
+    /// @param filter
+    ///   Serializable object representing the match criteria.
+    /// @param replacement
+    ///   The replacement document, as a serializable object.
+    /// @param options
+    ///   Optional arguments, see mongocxx::options::update.
+    ///
+    /// @return The result of attempting to replace a document.
+    /// @throws exception::write if the operation fails.
+    ///
+    /// @see http://docs.mongodb.org/manual/reference/command/update/
+    ///
+    mongocxx::stdx::optional<mongocxx::result::replace_one> replace_one(
```

Comment:
Added the following functions:

```
update_many(T filter, view_or_value update,options)
update_many(view_or_value filter, T update, options)
update_many(T filter, T update, options)
update_one(T filter, view_or_value update, options)
```

I feel like `update_one(T filter, T update, options)` would be pretty much just a `replace_one`, since the update parameter is just another object.